### DISCUSSION TITLE
How to upgrade code-server on railway.app?

### DISCUSSION THREAD

**Discussion by Slodziak190q** [coder/code-server#4066] [Q&A] [answered]
I got an pop-up that new uptade is available so how to install this uptade?

**Comment by lg-nineoneninethreenine** [comment, marked as answer]
Go to your GitHub repo where you hosted your Railway deployments. Open and edit the Dockerfile. In the first line, change `latest` to `3.11.1` and open the new Railway deployment. code-server is successfully updated.

**Reply by Slodziak190q** [reply]
Now all my files are gone how to get the files back?

**Reply by lg-nineoneninethreenine** [reply]
Since you used a Git repository, the files have to be there. If you didn't, I am sorry. I also have this bleeped-up issue, so I made a GitHub repository, linked it to Railway and used that. Note that you will have to push every changes to the repository and yes, it can be private.

**Reply by jsjoeio** [reply]
No, @im-coder-lg you are correct. That's how to update. 

@Slodziak190q if you want to persist files, you need to setup rclone. See [this](https://github.com/cdr/deploy-code-server/tree/main/deploy-container#-persist-your-filesystem-with-rclone)

**Reply by Slodziak190q** [reply]
Ok j got my files back
